src/controlCenter/widgets/mpris/mpris.vala

```diff
@@ -91,6 +92,24 @@ namespace SwayNotificationCenter.Widgets.Mpris {
                 // Clamp the radius
                 mpris_config.image_radius = mpris_config.image_radius.clamp (
                     0, (int) (mpris_config.image_size * 0.5));
+
+                // Blacklist
+                if (config.has_member("blacklist")) {
+                    Json.Object? blacklist_object = config.get_object_member("blacklist");
+                    if (blacklist_object != null) {
+                        var blacklisted_apps = new Gee.ArrayList<string>();
+                        foreach (var key in blacklist_object.get_members()) {
+                            Json.Object? app_object = blacklist_object.get_object_member(key);
+                            if (app_object != null && app_object.has_member("state") && app_object.has_member("app-name")) {
+                                string? state = app_object.get_string_member("state");
+                                if (state == "blacklisted") {
+                                    blacklisted_apps.add(app_object.get_string_member("app-name"));
+                                }
+                            }
+                        }
+                        mpris_config.blacklist = blacklisted_apps.to_array();
+                    }
+                }
```

Comment:
This can be greatly simplified by following how the other mpris config options are created (Using the ConfigModels logic to initialize and populate the blacklist option) :)

Reply:
Appreciate the well structured, constructive feedback.
I'll have another go at this one